### PR TITLE
PFW-1522 Implement filament change screen

### PR DESCRIPTION
**Pull request by gudnimg**
The pull request implements a new filament change screen for MMU users as proposed in PFW-1522. The screen is only shown when SpoolJoin is OFF. When SpoolJoin is ON, the printer will automatically try to load the next adjacent slot.

The screen behaves exactly like the new MMU error screens so I won't go into detailed describing that behavior here. The user can select `Eject` to remove any old filament (or just cut the tip of the filament to re-use it). **The screen will only be dismissed after clicking the Load button**. After pressing the `Load` button, the user is presented with a menu to select which slot to load from Filament 1 to Filament 5.

Screenshot of the fullscreen on MK3S+
====

* Default choice is `Load` (always the middle button)

Screenshot of the 'More' description fullscreen on MK3S+
====

* Clicking the knob will re-render the menu shown in the image above.

----

Note that since this is not an "error" -- similar to `EJECTED_FILAMENT` -- we need to make sure that the new error code `FILAMENT_CHANGE` does not increment any failure statistic counters.

----
- [x] Update the description when 'More' button is clicked to reflect https://github.com/prusa3d/Prusa-Error-Codes/pull/92/files
- [x] Load filament button should be the default choice (Middle button)
- [x] Make sure `FILAMENT_CHANGE` does not increment any failure statistics.
- [X] Update po files
- [x] Internal translators review
  - [x] CS @DRracer :heavy_check_mark: 
  - [x] DE @3d-gussner :heavy_check_mark: 
  - [x] ES
  - [x] FR
  - [x] IT @wavexx 
  - [x] PL
- [ ] Community translators review
  - [x] HR  @Prime1910 
  - [x] HU @AttilaSVK @Hauzman
  - [x] NL @3d-gussner :heavy_check_mark:  @vintagepc @stelgenhof
  - [x] NO @OS-kar @trondkla
  - [x] RO @leptun @Hauzman 
  - [x] SK @ingbrzy @DRracer :heavy_check_mark: 
  - [x] SV @Painkiller56 
- [X] _Merge PR https://github.com/prusa3d/Prusa-Error-Codes/pull/92_

Change in memory:
Flash: +152 bytes
SRAM: +1 byte

### CODE REVIEW

File: Firmware/mmu2.cpp
Comment (gudnimg) [inline Firmware/mmu2.cpp:728]: **Note:**
Right now I need this change, because `lastErrorCode` must be cleared. We cannot rely on checking the current error code from the MMU protocol since Printer errors do not originate from the protocol.

Consider the following scenario:

1.  `lastErrorCode` starts with value `ErrorCode::OK`
2. The firmware raises `FILAMENT_CHANGE` error. It's a new error code so `lastErrorCode = ErrorCode::FILAMENT_CHANGE`
3. The user clicks Load and the error screen is dismissed.
4. FINDA runout triggers M600 and the firmware raises `FILAMENT_CHANGE` error.
5. `lastErrorCode` is still `ErrorCode::FILAMENT_CHANGE`. This results in the error not being logged fully. Also implies `IncrementMMUFails()` would not be called if one got the same error twice e.g. failed load!

The purpose of `lastErrorCode` in my opinion is to prevent the error reporting to flood the serial log _while_ the error is unresolved. When the MMU reports an error, it will continue to keep reporting the same error every second. This we don't want in the logs. BUT, if the same error is resolved twice (e.g. failed load) I would expect to see two logged error events since they are independent. Also I would expect the statistics in the menus to reflect this as well, that is => counting two failures and not one.

Proposal: Set `lastErrorCode` to `ErrorCode::OK` when the error is resolved.

**Comment by gudnimg** [comment]
Found one bug. If I trigger M600 at boot-up with no filament loaded previously, then Ejecting filament can cause the firmware to hang endlessly as it will try to Eject filament "256" while the MMU is continously rejecting the command. We currently don't handle this scenario.

Will look into it now.

**Comment by gudnimg** [comment]
Rebased to sync with MK3 branch. There were some conflicts due to changes in the error code names.

**Comment by 3d-gussner** [comment]
@DRracer Please review it as you also commented on the https://github.com/prusa3d/Prusa-Error-Codes/pull/92 
Is PFW-1522 still happening?

**Comment by 3d-gussner** [comment]
Missing updates to po files

**Comment by 3d-gussner** [comment]
Tested on MK404 and works as described. I would approved that PR (with po files updated), waiting for @DRracer .

**Comment by 3d-gussner** [comment]
@gudnimg Updated and translated po files. See https://github.com/prusa3d/Prusa-Firmware/pull/4216/commits/31369f60fb477161d0060f7f076ce99efbb7dc70

We gonna need a review by the internal and community translators.

**Review by 3d-gussner — APPROVED** [review]
Tested and works as described.
Open tasks need to be done before merge.

**Comment by 3d-gussner** [comment]
@DRracer What is the status here?

**Comment by gudnimg** [comment]
Rebased to sync with MK3 branch

**Review by DRracer — APPROVED** [review]
Technically, I don't see anything wrong with this PR. I just have a strange feeling that we might have broken some scenario while refactoring `MMU2::CheckUserInput`. There, it has always been a bit brittle to handle _Load to extr. failed_ errors correctly in the past.

So, I'm approving this PR, but extensive tests need to be performed in this regard.

**Comment by gudnimg** [comment]
Rebased to resolve merge conflict

**Review by DRracer — APPROVED** [review]
Approving after rebase onto MK3 again.
We may need to improve some translations later on.

**Comment by 3d-gussner** [comment]
@leptun @Hauzman @AttilaSVK @Prime1910 @ingbrzy @shatter136 @Painkiller56 @OS-kar @trondkla

Hello, how are you? Please review the translations I prepared.

**Comment by AttilaSVK** [comment]
Hi!

Below are the corrections for the Hungarian translation.

#. MSG_BTN_EJECT c=8
#: ../../Firmware/mmu2/errors_list.h:357
../../Firmware/mmu2/errors_list.h:369
msgid "Eject"
msgstr "Kiadás"

#. MSG_TITLE_FILAMENT_CHANGE c=20
#: ../../Firmware/mmu2/errors_list.h:188
../../Firmware/mmu2/errors_list.h:235
msgid "FILAMENT CHANGE"
msgstr "FILAMENT CSERE"

#. MSG_BTN_LOAD c=8
#: ../../Firmware/mmu2/errors_list.h:356
../../Firmware/mmu2/errors_list.h:368
msgid "Load"
msgstr "Betöltés"

#. MSG_DESC_FILAMENT_CHANGE c=20 r=8
#: ../../Firmware/mmu2/errors_list.h:285
../../Firmware/mmu2/errors_list.h:340
msgid "M600 Filament Change. Load a new filament or eject the old one."
msgstr "M600 Filamentcsere. Helyezz be egy új filamentet, vagy vedd ki a
régit."

3d-gussner ***@***.***> ezt írta (időpont: 2023. aug. 15., K,
12:10):

File: lang/po/Firmware_ro.po
Comment (leptun) [inline lang/po/Firmware_ro.po]: "SCHIMBARE FILAMENT."

File: lang/po/Firmware_ro.po
Comment (leptun) [inline lang/po/Firmware_ro.po]: "M600: Schimbare filament. Încărcați un filament nou sau scoateți-l pe cel vechi.

**Comment by Prime1910** [comment]
Croatian translation, line 2567 
"Promjena žarne niti M600. Stavite novu nit ili izbacite staru." => "Promjena filamenta M600. Stavite novu nit ili izbacite staru."

File: lang/po/Firmware_ro.po
Comment (Ro3Deee) [reply lang/po/Firmware_ro.po]: Or "M600: Schimbare filament. Încarcă un filament nou sau scoate-l pe cel vechi."
Less used memory this way

**Comment by Hauzman** [comment]
Ro as leptun suggested will be ok

**Review by shatter136 — COMMENTED** [review]
SK looks OK asi is, maybe just put "ť" instead of "t" in line 2550, so that it will be "Zaviesť"

**Comment by stelgenhof** [comment]
Dutch translation (NL) is correct and looks good to me.

**Comment by DRracer** [comment]
@AttilaSVK @leptun @petrubecheru @Prime1910 @Hauzman @shatter136 @stelgenhof thank you, translations updated

**Comment by DRracer** [comment]
@Painkiller56 @OS-kar @trondkla :pray: could you please have a look at the NO and SV translations in this PR? Thank you.

**Review by trondkla — APPROVED** [review]
NO language looks -good- OK, one change maybe on Norwegian

File: lang/po/Firmware_no.po
Comment (trondkla) [inline lang/po/Firmware_no.po]: Would "Last inn" fit? I think that would be a little bit better.

Comment (trondkla) [reply lang/po/Firmware_no.po]: @DRracer If not maybe "Last" is just good enough?
"Laste" is infinitive, so it's "to load", so I don't think it's correct :) But people would understand it

Comment (DRracer) [reply lang/po/Firmware_no.po]: "Last inn" would fit, but we'll have issues with the space. Could we do something like "Last-inn"? Just to prevent space in the button's name.

Comment (trondkla) [reply lang/po/Firmware_no.po]: The usage of - is uncommon in Norwegian.

In this context, where you have "Løs ut" (eject) as another alternative, "Last" should be enough.
In norwegian you can say load in (load) and load out (unload), but since you have the context it's ok.

I tried to check if we could change eject to make that only a one word-no space-button, but "Utløs" is more to "trigger" then ejecting something.

So I think "Løs ut" and "Last" is the best short forms we can have, and "Last Inn", while being more correct, "Last-Inn" looks really weird, but maybe weird is better than potensiel ambiguous?

Comment (trondkla) [reply lang/po/Firmware_no.po]: Personally both "Last" and "Last-Inn" works for me, but only Last is technically correct, however ambiguous without a context.

Comment (3d-gussner) [reply lang/po/Firmware_no.po]: @trondkla Thanks as the limit is 8 characters it should fit
```
01234567
Eject
Løs ut

01234567
Load
Last-inn
```

Comment (3d-gussner) [reply lang/po/Firmware_no.po]: @trondkla Do you know by chance bit Swedish as we didn't get an answer yet from the SV translation owner

Comment (trondkla) [reply lang/po/Firmware_no.po]: I understand and can read swedish, but I do not speak fluent.
It looks good based on the other Swedish translations.

**Review by DRracer — APPROVED** [review]

**Pull request merged**
